src/Components/Browser.JS/package.json

```diff
@@ -5,14 +5,16 @@
   "description": "",
   "main": "index.js",
   "scripts": {
+    "preclean": "cd utils && yarn install --mutex network",
+    "clean": "node utils/node_modules/rimraf/bin.js node_modules",
+    "prebuild": "yarn run clean && yarn install --mutex network",
     "build": "yarn run build:debug && yarn run build:production",
     "build:debug": "cd src && webpack --mode development --config ./webpack.config.js",
     "build:production": "cd src && webpack --mode production --config ./webpack.config.js",
     "test": "jest"
   },
   "devDependencies": {
-    "@aspnet/signalr": "^1.0.0",
-    "@aspnet/signalr-protocol-msgpack": "^1.0.0",
+    "@aspnet/signalr-protocol-msgpack": "file:../../SignalR/clients/ts/signalr-protocol-msgpack",
```

Comment:
Would this be problematic if we choose to publish the Blazor package to NPM? When someone references our package, where is NPM going to pull this dependency from?

NPM has other (but less convenient) ways of doing source-to-source references during development that interact properly with package references in production, i.e., `npm link`. I'm not saying that's necessarily what we need to do right now, but let's be sure this is not creating a problem for ourselves.

Reply:
OK, I see from another comment lower down that there's some kind of `package.json`-rewriting trickery going on to deal with this. Having so much customization over the build process is a worrisome trap we keep falling into, but I don't necessarily have a better suggestion in this case.

If you could at least confirm that the rewriting is what avoids this problem that would be good to know!